c/swig/typedb_driver_csharp.swg

```diff
@@ -0,0 +1,683 @@
+/*
+ * Copyright (C) 2022 Vaticle
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+%csconst(1);
+
+%typemap(cscode) SWIGTYPE %{
+    public $csclassname Released()
+    {
+        var cPtr = swigCPtr.Handle;
+        if (swigCMemOwn)
```

Comment:
Didn't think about it! I can wrap the body of this method in `lock(this) {}` just as the SWIG-generated `Dispose` does it just not to worry about it.

Reply:
Well, not every place, just `Dispose`. `swigRelease` does not, that's why I am unsure.

Reply:
In that case `Release()` twice without a lock doesn't introduce extra races, because those were already present for all native calls.
Illustration for why the lock wouldn't help here: https://dotnetfiddle.net/kO2zJf

Reply:
Well, yeah, you're right about the races and locks, looks like it already works this way, I guess...